qiskit_experiments/curve_analysis/curve_analysis.py

```diff
@@ -915,7 +824,10 @@ def _run_analysis(
                     analysis_results.append(result_entry)
 
             # add extra database entries
-            analysis_results.extend(self._extra_database_entry(fit_result))
+            device_components = [
```

Comment:
Can you turn this into method call so that some experiment can customize. Currently Readout spectroscopy does this in base analysis class hook (so probably we can remove it).

Can you also update these lines?

1. all parameter entry:
https://github.com/Qiskit/qiskit-experiments/blob/f612c0b4ce07e1a38e1e259ab5b3609083cc35e7/qiskit_experiments/curve_analysis/curve_analysis.py#L873-L885

2. special parameter entry:
https://github.com/Qiskit/qiskit-experiments/blob/f612c0b4ce07e1a38e1e259ab5b3609083cc35e7/qiskit_experiments/curve_analysis/curve_analysis.py#L908-L914

3. data points entry (hopefully this will be a part of artifact):
https://github.com/Qiskit/qiskit-experiments/blob/f612c0b4ce07e1a38e1e259ab5b3609083cc35e7/qiskit_experiments/curve_analysis/curve_analysis.py#L930-L937

Reply:
fyi
https://github.com/Qiskit/qiskit-experiments/blob/f612c0b4ce07e1a38e1e259ab5b3609083cc35e7/qiskit_experiments/library/characterization/analysis/resonator_spectroscopy_analysis.py#L40-L42